tests/integration/pkg/testcontext/config_validation.go

```diff
@@ -0,0 +1,98 @@
+package testcontext
+
+import (
+	"fmt"
+	"github.com/kyma-project/api-gateway/tests/integration/pkg/helpers"
+	"github.com/kyma-project/api-gateway/tests/integration/pkg/resource"
+	"k8s.io/client-go/dynamic"
+	"log"
+	"strings"
+)
+
+const localDomain = "local.kyma.dev"
+
+func (c *Config) RequireDomain() error {
+	if c.Domain == "" {
+		return fmt.Errorf("test domain is required")
+	}
+	return nil
+}
+
+func (c *Config) RequireCustomDomain() error {
+	if c.CustomDomain == "" {
+		return fmt.Errorf("custom domain is required")
+	}
+	return nil
+}
+
+func (c *Config) RequireCredsForOIDC() error {
+	if c.OIDCConfigUrl != "" {
+		if c.ClientID == "" {
+			return fmt.Errorf("client ID is required")
+		}
+		if c.ClientSecret == "" {
+			return fmt.Errorf("client Secret is required")
+		}
+	}
+	return nil
+}
+
+func (c *Config) ValidateGardener(resourceMgr *resource.Manager, k8sClient dynamic.Interface) error {
+	if c.IsGardener {
+		err := helpers.CheckGardenerSetup(resourceMgr, k8sClient)
+		if err != nil {
+			return fmt.Errorf("requested Gardener cluster, but the current cluster is not properly installed: %w", err)
+		}
+		if strings.Contains(c.Domain, localDomain) {
```

Comment:
What if the `c.Domain` is set to something else than `local.kyma.dev`? For example `asd.example`. Is there a way to ensure in this part of the code, that the domain is Gardener's?

Reply:
We are considering only tests here. So there are only few cases possible:
- locally run tests, so everything works on localhost, that is why we have *.local.kyma.dev pointing to 127.0.0.1
- tests  run against the Gardener cluster using default cluster domain, which is stored in shoot info, so we validate whether they match
- some custom setup with custom domain - but this is done in custom-domain tests and then the additional domain is provided as a separate configuration attribute and not validated in any way, because it is completely different than the default Gardener one

Reply:
Okay, so we decided to change the logic here and auto detect the Gardener cluster, so the validation is useful also when an end-developer doesn't set IS_GARDENER=true